api/pointer_utils.go

```diff
@@ -59,7 +59,9 @@ func bytePtr(x []byte) *[]byte {
 	// Don't return if it's all zero.
 	for _, v := range x {
 		if v != 0 {
-			return &x
+			xx := make([]byte, len(x))
+			copy(xx, x)
+			return &xx
```

Comment:
Could you also update `postgres.go : baPtr`?
```
func baPtr(x []byte) *[]byte {
	if x == nil || len(x) == 0 {
		return nil
	}
	allzero := true
	for _, b := range x {
		if b != 0 {
			allzero = false
			break
		}
	}
	if allzero {
		return nil
	}
	out := new([]byte)
	*out = x
	return out
}
```

Reply:
Is there a difference between `bytePtr()` and `baPtr()`? I can just delete the later.

Reply:
Just that they're private functions in different packages, I think there are a few cases where we have duplicate functions for the pointer utilities.

Reply:
It would be nice to have only one function in `util/`, but I changed `baPtr()` for now.